### PR TITLE
Types ValidateBasic Tests

### PR DESCRIPTION
**Pull request by alexanderbez**
## Description

Increase test coverage for `ValidateBasic` for the following types: `Header`, `CommitSig` and `Commit`.

Closes: #5416

### CODE REVIEW

File: types/block_test.go [hunk 1]
```diff
@@ -865,3 +865,447 @@ func TestBlockIDEquals(t *testing.T) {
...
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			err := tc.commit.ValidateBasic()
+			require.Equal(t, tc.expectErr, err != nil, err)
```
Comment (melekes) [inline types/block_test.go:1308]: I always thought this is a very brittle way to assert the error. We don't check the specific error here. If someone makes an error in the code and we get, for example, "invalid block ID" while expecting "invalid signature", the test will still pass, but we won't know that we've made an error.

Now, I don't know how critical this is in this context, but definitely, something to keep in mind.

**Review by melekes — APPROVED** [review]
👍

File: types/block_test.go (hunk 1 above)
Comment (alexanderbez) [reply types/block_test.go:1308]: This is true, but for many of these we don't return sentinel errors. Do you propose we do a substring match?

Comment (melekes) [reply types/block_test.go:1308]: > Do you propose we do a substring match?

yep

**Pull request merged**